src/Piral.Blazor.Tools/content/src/index.tsx

```diff
@@ -3,6 +3,10 @@ import * as Blazor from './blazor.codegen';
 import './**BlazorProjectName**.styles.css';
 
 export function setup(app: PiletApi) {
+    const pb = Blazor.getPiralBlazorApi(app);
+    app.fromBlazor = pb.fromBlazor;
```

Comment:
This should not be set / replaced in case of the standard app-shell hosted model.

Why don't you just have this code in the `getPiralBlazorApi` function?

So it would look like this:

```ts
export function setup(app: PiletApi) {
  Blazor.getPiralBlazorApi(app);
  Blazor.registerDependencies(app);
  // ...
}
```

Reply:
You could rename to `Blazor.initBlazorApi(app);`